File: solo-leveling-ui-service/src/main/java/com/sleepkqq/sololeveling/ui/model/UserRole.java
package com.sleepkqq.sololeveling.ui.model;

import org.springframework.security.core.GrantedAuthority;

public enum UserRole implements GrantedAuthority {
  USER,
  ADMIN;

  @Override
  public String getAuthority() {
    return name();
  }

  public boolean isUser() {
    return this == USER || isAdmin();
  }

  public boolean isAdmin() {
    return this == ADMIN;
  }
}

File: solo-leveling-ui-service/src/main/kotlin/com.sleepkqq.sololeveling/ui/model/UserRole.kt
package com.sleepkqq.sololeveling.ui.model

import org.springframework.security.core.GrantedAuthority

enum class UserRole : GrantedAuthority {
	USER,
	ADMIN;

	override fun getAuthority(): String = name

	val isUser: Boolean
		get() = this == USER || isAdmin

	val isAdmin: Boolean
		get() = this == ADMIN
}